### PR TITLE
GMP Alerts: HAProxy

### PR DESCRIPTION
**Pull request by jsirianni**
Ported alerts from [cloud monitoring](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/tree/master/alerts/haproxy-gke) to monitoring.googleapis.com/v1 Rules.

- Added alerts configuration for HAProxy
- Updated HAProxy documentation.yaml to point to new alerts configuration

Let me know if `haproxy_alerts.yaml` should exist in a different directory. For now, I have kept it with the GMP metadata yamls.

### CODE REVIEW

**Comment by jsirianni** [comment]
With the rules modified to trigger, I see alerts in alert manager.

Screenshot from 2022-11-17 12-49-35

File: integrations/haproxy/haproxy_alerts.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+apiVersion: monitoring.googleapis.com/v1
+kind: Rules
+metadata:
+  name: rabbitmq-rules
+  labels:
+    app.kubernetes.io/component: rules
+    app.kubernetes.io/name: rabbitmq-rules
```
Comment (yqlu) [inline integrations/haproxy/haproxy_alerts.yaml:7]: rabbitmq --> haproxy throughout this file?

File: integrations/haproxy/haproxy_alerts.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    app.kubernetes.io/part-of: google-cloud-managed-prometheus
+spec:
+  groups:
+  - name: rabbitmq
+    interval: 30s
+    rules:
+    - alert: RedisDown
```
Comment (yqlu) [inline integrations/haproxy/haproxy_alerts.yaml:14]: Redis --> HAProxy

**Review by yqlu — COMMENTED** [review]
Just a couple of copy-paste issues. 

Keeping the file in the same directory is fine -- we'll move it around as needed, thanks!

File: integrations/haproxy/haproxy_alerts.yaml (hunk 1 above)
Comment (jsirianni) [reply integrations/haproxy/haproxy_alerts.yaml]: Of course. Started with rabbit but pivoted to haproxy :facepalm:

**Review by yqlu — APPROVED** [review]

File: integrations/haproxy/documentation.yaml [hunk 3]
```diff
@@ -83,3 +83,5 @@ podmonitoring_config: |
       matchLabels:
         app.kubernetes.io/name: haproxy
 sample_promql_query: up{job="haproxy", cluster="{{cluster_name}}", namespace="{{namespace_name}}"}
+alerts_config: {% includecode content_path="stackdriver/docs/managed-prometheus/exporters/configs/haproxy_alerts.yaml" %}
```
Comment (EvanSimpson) [inline integrations/haproxy/documentation.yaml:86]: I'm worried about having a file called _alerts.yaml outside of the top-level alerts directory of this repo (meant to contain templates for Cloud alerting, not prometheus). Would it be possible to embed the file content here as a string?

Reading around online it should be possible using the `|` yaml string delimiter and some additional indentation. i.e. https://medium.com/@george.shuklin/embedding-yaml-into-yaml-3511246d7ea7

Comment (jsirianni) [reply integrations/haproxy/documentation.yaml]: Not a problem, done.

Comment (EvanSimpson) [reply integrations/haproxy/documentation.yaml]: Great, thank you!

**Review by EvanSimpson — APPROVED** [review]

**Pull request merged**